lib/Service/MLP/OptimizerService.php

```diff
@@ -80,12 +81,29 @@ private function printConfig(int $epoch,
 	}
 
 	/**
-	 * @param Model ...$models
+	 * @param OutputInterface $output
+	 * @param TrainingResult ...$results
 	 */
-	private function getAverageCost(Model ...$models): float {
-		$costs = array_map(function (Model $model) {
-			return ($model->getPrecisionN() + $model->getRecallN()) / 2;
-		}, $models);
+	private function getAverageCost(OutputInterface $output,
+									TrainingResult ...$results): float {
+		$costs = array_map(function (TrainingResult $result) use ($output) {
+			$output->writeln(sprintf("  Training result: f1=%f, p(n)=%f, r(n)=%f, f1(n)=%f, p(y)=%f, r(y)=%f, f1(y)=%f, PSR=%d/%d/%d",
```

Comment:
Not a strong opinion there but isn't it more readable when the first parameter is on a new line and indented like the others?

Reply:
![Bildschirmfoto von 2021-01-29 14-16-48](https://user-images.githubusercontent.com/1374172/106279393-a8885980-623c-11eb-8877-e6c8b0fb3355.png)

There is a lot of info to show with the many epochs, so I try to have the info as dense as possible